### DISCUSSION TITLE
.net 8.0 Service: Do not call GetAuthenticationStateAsync outside of the DI scope for a Razor component

### DISCUSSION THREAD

**Discussion by jmpotvin** [MarimerLLC/csla#4088] [Questions] [answered]
I am working with Scoped Services that retrieve information using CSLA Business Objects. After upgrading to version 8.2.5, I encountered the following error:

`Do not call GetAuthenticationStateAsync outside of the DI scope for a Razor component `

I noticed that this issue might have been addressed in #3614 , but it still persists for me with the following configuration:

```
builder.Services.AddCsla(o => o
             .AddAspNetCore()
             .AddServerSideBlazor(ssb => ssb.UseInMemoryApplicationContextManager = true)
             .Security(so => so.FlowSecurityPrincipalFromClient = true)
             .DataPortal(dpo => dpo
               .AddServerSideDataPortal()
               .ClientSideDataPortal(co => co
                   .UseLocalProxy())));
```              

Can someone provide guidance on the proper configuration needed to call a CSLA Business Object from a Service within a Razor component?

**Comment by rockfordlhotka** [comment, marked as answer]
Hmm. The issue in #3614 was with `UseInMemoryApplicationContextManager = false`. I didn't retest it with this configuration, but I guess I should have 😔
